### PR TITLE
Added libcddb (full build including regex and libcdio)

### PR DESCRIPTION
**Pull request by nschlia**
Added libcddb support (32/64 bit shared and static version)

### CODE REVIEW

**Comment by nschlia** [comment]
This is the libcddb package, including required fixes. I hope I made it correctly. If not, just drop me a note. I also want to add libmusicbrainz, libcoverart and libneon. I'll do it as soon as I know I am doing it right.

File: src/libcddb-1-windows-macro-fix.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+This file is part of MXE. See LICENSE.md for licensing information.
+
+Newer versions winioctl.h contain a #define SEARCH_ALL .. which collides with the enum in cddb_conn.h.
```
Comment (starius) [inline src/libcddb-1-windows-macro-fix.patch]: Please format the patch with patch-tool-mxe.
See https://github.com/mxe/mxe/pull/1277#issuecomment-257083463 for more info on it.

File: src/libcddb.mk [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# mxe/src/libcddb.mk
+# This file is part of MXE.
+# See index.html for further information.
```
Comment (starius) [inline src/libcddb.mk]: Replace the header with
```
# This file is part of MXE. See LICENSE.md for licensing information.
```

File: src/libcddb.mk [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+PKG             := libcddb
+$(PKG)_IGNORE   := 
+$(PKG)_VERSION  := 1.3.2
+$(PKG)_CHECKSUM := 35ce0ee1741ea38def304ddfe84a958901413aa829698357f0bee5bb8f0a223b
+$(PKG)_SUBDIR   := $(PKG)-$($(PKG)_VERSION)
+$(PKG)_FILE     := $(PKG)-$($(PKG)_VERSION).tar.bz2
+$(PKG)_URL      := http://downloads.sourceforge.net/project/libcddb/libcddb/$($(PKG)_VERSION)/$(PKG)-$($(PKG)_VERSION).tar.bz2
```
Comment (starius) [inline src/libcddb.mk]: Use `$(SOURCEFORGE_MIRROR)` here.

File: src/libcddb.mk [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# ac_cv_func_malloc_0_nonnull=yes		avoid unresolved external
+# ac_cv_func_realloc_0_nonnull=yes		avoid unresolved external
+define $(PKG)_BUILD
+    cd '$(1)' && \
+	lt_cv_deplibs_check_method="pass_all" \
+    	ac_cv_func_malloc_0_nonnull=yes \
+    	ac_cv_func_realloc_0_nonnull=yes \
+    	LDFLAGS="-L$(PREFIX)/$(TARGET)/lib/ -L$(PREFIX)/$(TARGET)/bin/" \
+    	CPPFLAGS="-I$(PREFIX)/$(TARGET)/include/" \
+    	./configure \
```
Comment (starius) [inline src/libcddb.mk]: Use out-of-tree builds (grep by `SOURCE_DIR` and `BUILD_DIR` to find examples).

**Comment by starius** [comment]
Can you add small sample programs and compile them as a part of build, please? They are needed to check for linking errors. Write output binary to `usr/<target>/bin/test-<pkg>.exe`

See `src/lensfun.mk` for an example.

**Comment by starius** [comment]
Failed to build libcddb: https://gist.github.com/c2bc0367aa39a125e51287f1fb2ab076

**Comment by starius** [comment]
Failed to build libdvdetect: https://gist.github.com/34f79f82534b5599b0f968e9a6a82957
Add `tinyxml` to its dependencies.
But after that static targets are broken with linking errors: https://gist.github.com/65a0253b1d55596aee985e411b2b77dc
Some libraries should be added to linker options.

Shared targets pass after installing tinyxml, but they have another problem. Many tools are installed to `bin/`:

```
#       i686-w64-mingw32.shared/bin/dvdinfo.exe
#       i686-w64-mingw32.shared/bin/dvdinfocxx.exe
#       i686-w64-mingw32.shared/bin/dvdlookup.exe
#       i686-w64-mingw32.shared/bin/dvdlookupcxx.exe
#       i686-w64-mingw32.shared/bin/dvdsubmit.exe
#       i686-w64-mingw32.shared/bin/dvdsubmitcxx.exe
```

Core MXE packages (src/*.mk) are libraries and they should not install tools. Add `$(MXE_DISABLE_PROGRAMS)` to options of `make` to turn off installation of tools. If you want to install these tools, add them as a package to plugins/apps. See https://github.com/mxe/mxe/tree/master/plugins for instructions.

**Comment by starius** [comment]
neon builds successfully. There are two minor problems though:

1. It installs many documentation and man files. Please disable by adding `$(MXE_DISABLE_DOCS)` to `make`.
2. Can you elaborate on purpose of file `<target>/bin/neon-config`, please? It is a shell script. Is it needed in MXE setup?

**Comment by nschlia** [comment]
Wow, did expect issues, but not so many... I'll fix them ASAP. Thanks!

**Comment by nschlia** [comment]
libdvdetect has been removed for now, it was added accidentally. It was not ready to be included in mxe. I will do that later.
neon was also added accidentally, but as it passed the tests, OK, why not :)
I still have to add usr/<target>/bin/test-<pkg>.exe tests yet, everything else should be done. For cddb there's a cddb_query.exe that could be used.

Question:

I guess I'll have to close the request when I am ready (to also avoid getting more changes from my repo?). 
Sorry, I am good in svn but git keeps giving me the creeps. I'll need to work with it ahwile to get a better grip on it.

**Comment by starius** [comment]
> I guess I'll have to close the request when I am ready (to also avoid getting more changes from my repo?).

Independent things are better to be done in separate pull requests. They are easier to review and to update. You do not have to do them one-by-one, because multiple pull requests can be done in parallel.

You can split this pull request to two by making new branches (`git checkout -b new-branch-name`), modifying the history (e.g. with `git rebase --interactive`) and `git push -f` to update the branch on Github.

> Sorry, I am good in svn but git keeps giving me the creeps. I'll need to work with it ahwile to get a better grip on it.

Github pull request corresponds to git branch. You can read about [git branches](https://www.digitalocean.com/community/tutorials/how-to-use-git-branches) and [git rebase](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request).

**Comment by nschlia** [comment]
Now all should be fixed. I compiled all shared/static libs properly.

**Comment by starius** [comment]
Why have you closed the pull request?

**Comment by nschlia** [comment]
Wrong button. Silly me :)

**Comment by nschlia** [comment]
This is it. My nightly build went through without issues. You may have a look at it anytime now.

File: src/libcddb.mk [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)'
+    $(MAKE) -C '$(BUILD_DIR)' -j 1 install
+
+    # Test binary    
+    mv '$(PREFIX)/$(TARGET)/bin/cddb_query.exe' '$(PREFIX)/$(TARGET)/bin/test-$(PKG).exe'
+endef
+
+$(PKG)_BUILD_i686-pc-mingw32    = $(subst @special-target@, x86-win32-gcc,    $($(PKG)_BUILD))
+$(PKG)_BUILD_i686-w64-mingw32   = $(subst @special-target@, x86-win32-gcc,    $($(PKG)_BUILD))
+$(PKG)_BUILD_x86_64-w64-mingw32 = $(subst @special-target@, x86_64-win64-gcc, $($(PKG)_BUILD))
```
Comment (starius) [inline src/libcddb.mk]: Why these three lines are needed?

File: src/libdvdetect.mk [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        $(MXE_CONFIGURE_OPTS)
+    $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)'
+    $(MAKE) -C '$(BUILD_DIR)/include' -j '$(JOBS)' install
+    $(MAKE) -C '$(BUILD_DIR)/lib' -j '$(JOBS)' install
+endef
+
+$(PKG)_BUILD_i686-pc-mingw32    = $(subst @special-target@, x86-win32-gcc,    $($(PKG)_BUILD))
+$(PKG)_BUILD_i686-w64-mingw32   = $(subst @special-target@, x86-win32-gcc,    $($(PKG)_BUILD))
+$(PKG)_BUILD_x86_64-w64-mingw32 = $(subst @special-target@, x86_64-win64-gcc, $($(PKG)_BUILD))
+
```
Comment (starius) [inline src/libdvdetect.mk]: Style: remove trailing empty lines from mk-files, please.

File: src/neon.mk [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        ne_cv_fmt_size_t=%lu \
+        ne_cv_fmt_ssize_t=%lu \
+        ne_cv_fmt_off64_t=%I64u \
+        ne_cv_fmt_time_t=%lu \
+        ne_cv_libsfor_socket=-lws2_32 \
+        ne_cv_libsfor_gethostbyname=-lws2_32 \
+    	'$(SOURCE_DIR)'/configure \
+        $(MXE_CONFIGURE_OPTS) \
+        $(MXE_DISABLE_DOCS) \
+	--with-ssl=yes
```
Comment (starius) [inline src/neon.mk]: Style: could you replace all tabs with spaces in mk-files, please.
Indent arguments of ./configure deeper.

File: src/libdvdetect.mk [hunk 8]
```diff
@@ -0,0 +1,32 @@
+# This file is part of MXE. See LICENSE.md for licensing information.
+
+PKG             := libdvdetect
+$(PKG)_IGNORE   := 
+$(PKG)_VERSION  := 0.71.0
+$(PKG)_CHECKSUM := b098e04660532df78836f50bc0a8044b66c6659b07a6bff6609724ad30a87192
+$(PKG)_SUBDIR   := $(PKG)-$($(PKG)_VERSION)
+$(PKG)_FILE     := $(PKG)-$($(PKG)_VERSION).tar.gz
+$(PKG)_URL      := https://github.com/nschlia/libdvdetect/releases/download/RELEASE_0_71/$(PKG)-$($(PKG)_VERSION).tar.gz
+$(PKG)_DEPS     := gcc tinyxml
```
Comment (starius) [inline src/libdvdetect.mk]: (Not to fix in this pull request.)
The build prints `tinyxml is deprecated - please use tinyxml2`
Upgrading to tinyxml2 requires code changes in libdvdetect.
I think, it makes sense to do it in https://github.com/nschlia/libdvdetect and then update MXE version of libdvdetect replacing tinyxml dependency with tinyxml2.

File: src/libdvdetect.mk [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    tail -1
+endef
+
+define $(PKG)_BUILD
+    cd '$(BUILD_DIR)' && \
+    	'$(SOURCE_DIR)'/configure \
+        $(MXE_CONFIGURE_OPTS)
+    $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)'
+    $(MAKE) -C '$(BUILD_DIR)/include' -j '$(JOBS)' install
+    $(MAKE) -C '$(BUILD_DIR)/lib' -j '$(JOBS)' install
```
Comment (starius) [inline src/libdvdetect.mk]: libdvdetect does not install pc file and test. Can you fix this, please?

File: src/neon.mk [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        ne_cv_fmt_off64_t=%I64u \
+        ne_cv_fmt_time_t=%lu \
+        ne_cv_libsfor_socket=-lws2_32 \
+        ne_cv_libsfor_gethostbyname=-lws2_32 \
+    	'$(SOURCE_DIR)'/configure \
+        $(MXE_CONFIGURE_OPTS) \
+        $(MXE_DISABLE_DOCS) \
+	--with-ssl=yes 
+    $(MAKE) -C '$(BUILD_DIR)' -j '$(JOBS)'                                                                                          
+    $(MAKE) -C '$(BUILD_DIR)' -j 1 install-lib install-headers install-nls
```
Comment (starius) [inline src/neon.mk]: neon does not install pc file and test. Can you fix this, please?

File: src/libcddb.mk (hunk 5 above)
Comment (nschlia) [inline src/libcddb.mk]: Not as far as I know. I copied them from the template. Ifg you think they are not required I can remove them. I actually do not know what they are for...

File: src/libdvdetect.mk (hunk 8 above)
Comment (nschlia) [inline src/libdvdetect.mk]: Yup. This is on my list, but will be done later (requires some testing, of course).

File: src/libdvdetect.mk (hunk 9 above)
Comment (nschlia) [inline src/libdvdetect.mk]: You mean it does not use pkgconfig? I could create the files "manually" as in xxHash.mk. Would that be OK?

File: src/neon.mk (hunk 10 above)
Comment (nschlia) [inline src/neon.mk]: pc files should be no problem, but actually I do not know how to use neon (it is used in a 3rd party library I use, not directly on my code). Is not having a test really a no-go?

File: src/libcddb.mk (hunk 5 above)
Comment (pavelvat) [inline src/libcddb.mk]: @starius because https://github.com/mxe/mxe/wiki/Add-a-New-Package:-Full-Version

File: src/libdvdetect.mk (hunk 9 above)
Comment (pavelvat) [inline src/libdvdetect.mk]: For example how to create correct pc file see:
https://github.com/mxe/mxe/blob/master/src/cryptopp.mk

Comment (nschlia) [inline src/libdvdetect.mk]: @pavelvat: Thanks. Your hint got me what I needed to know.

File: src/libcddb.mk (hunk 5 above)
Comment (nschlia) [inline src/libcddb.mk]: Removed the lines.

**Comment by nschlia** [comment]
All done besides of the libneon test binary. I could make one from an example I found but I would rather refrain from that if its OK.

File: src/libdvdetect.mk (hunk 8 above)
Comment (starius) [inline src/libdvdetect.mk]: Ok

File: src/libdvdetect.mk [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+     echo ''; \
+     echo 'Name: $(PKG)'; \
+     echo 'Version: $($(PKG)_VERSION)'; \
+     echo 'Description: fast database lookup for DVDs'; \
+     echo 'Libs: -L$${libdir} -ldvdetect'; \
+     echo 'Cflags: -I$${includedir}';) \
+     > '$(PREFIX)/$(TARGET)/lib/pkgconfig/$(PKG).pc'
+
+    # copy a test binary
+    cp '$(BUILD_DIR)/examples/c/.libs/dvdinfo.exe' '$(PREFIX)/$(TARGET)/bin/test-$(PKG).exe'
```
Comment (starius) [inline src/libdvdetect.mk]: This fails here: https://gist.github.com/d6c7572bae0a363e1b6672a41dd53b4f

File: src/libcddb-1-fixes.patch [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+index 1111111..2222222 100644
+--- a/include/cddb/cddb_conn.h
++++ b/include/cddb/cddb_conn.h
+@@ -59,6 +59,10 @@ typedef struct cddb_conn_s cddb_conn_t;
+  * Which fields to use for the full text search is defined by one or
+  * more of the constants below.
+  */
++#ifdef SEARCH_ALL
++#undef SEARCH_ALL   // Defined in winioctl.h, causes havoc under windoze
++#endif
```
Comment (starius) [inline src/libcddb-1-fixes.patch]: (Not in this PR.)
Can you fix / report this to upstream, please? If this symbol is renamed to SEARCH_EVERYTHING (or other name) in next release of libcddb, we could remove this patch.

File: src/libcddb-1-fixes.patch [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+@@ -100,6 +100,13 @@ cddb_conn_t *cddb_new(void)
+         c->use_cache = CACHE_ON;
+         /* construct cache dir '$HOME/[DEFAULT_CACHE]' */
+         s = getenv("HOME");
++        if (s == NULL)
++            s = getenv("TEMP");
++        if (s == NULL)
++            s = getenv("TMP");
++        if (s == NULL)
++			s = ".";
```
Comment (starius) [inline src/libcddb-1-fixes.patch]: Replace tab with spaces for consistency.

File: src/libcddb.mk [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+# lt_cv_deplibs_check_method="pass_all"        allow all libs (avoid static lib creation for x64 because of ws2_32.lib)
+# ac_cv_func_malloc_0_nonnull=yes        avoid unresolved external
+# ac_cv_func_realloc_0_nonnull=yes        avoid unresolved external
+define $(PKG)_BUILD
+   cd '$(BUILD_DIR)' && \
+    lt_cv_deplibs_check_method="pass_all" \
+        ac_cv_func_malloc_0_nonnull=yes \
+        ac_cv_func_realloc_0_nonnull=yes \
+        LDFLAGS="-L$(PREFIX)/$(TARGET)/lib/ -L$(PREFIX)/$(TARGET)/bin/" \
+        CPPFLAGS="-I$(PREFIX)/$(TARGET)/include/" \
```
Comment (starius) [inline src/libcddb.mk]: Redefining LDFLAGS and CPPFLAGS seem unneeded. I removed these two lines and it worked fine.

MXE_CONFIGURE_OPTS does its job :)

File: src/libcddb.mk [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+        '$(SOURCE_DIR)'/configure \
+        $(MXE_CONFIGURE_OPTS)
```
Comment (starius) [inline src/libcddb.mk]: Indent

File: src/libcddb.mk [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+    sort | uniq | \
+    head -1
+endef
+
+# lt_cv_deplibs_check_method="pass_all"        allow all libs (avoid static lib creation for x64 because of ws2_32.lib)
+# ac_cv_func_malloc_0_nonnull=yes        avoid unresolved external
+# ac_cv_func_realloc_0_nonnull=yes        avoid unresolved external
+define $(PKG)_BUILD
+   cd '$(BUILD_DIR)' && \
+    lt_cv_deplibs_check_method="pass_all" \
```
Comment (starius) [inline src/libcddb.mk]: Indent 4 spaces right.

File: src/libcddb.mk [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+    $(SED) -n 's,.*libcddb-\([0-9][^>]*\)\.tar.*,\1,p' | \
+    sort | uniq | \
+    head -1
+endef
+
+# lt_cv_deplibs_check_method="pass_all"        allow all libs (avoid static lib creation for x64 because of ws2_32.lib)
+# ac_cv_func_malloc_0_nonnull=yes        avoid unresolved external
+# ac_cv_func_realloc_0_nonnull=yes        avoid unresolved external
+define $(PKG)_BUILD
+   cd '$(BUILD_DIR)' && \
```
Comment (starius) [inline src/libcddb.mk]: Indent 1 space right.

File: src/libdvdetect.mk [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+    grep '<a href="/nschlia/libdvdetect/archive/' | \
+    $(SED) -n 's,.*href="/nschlia/libdvdetect/archive/RELEASE_\([0-9][^"]*\)\.tar.*,\1,p' | \
+    sort | uniq | \
+    tail -1
+endef
+
+define $(PKG)_BUILD
+    cd '$(BUILD_DIR)' && \
+              '$(SOURCE_DIR)'/configure \
+              $(MXE_CONFIGURE_OPTS)
```
Comment (starius) [inline src/libdvdetect.mk]: Unindent  left 6 spaces.

File: src/libcddb-1-fixes.patch (hunk 12 above)
Comment (nschlia) [inline src/libcddb-1-fixes.patch]: Don't think this is possible, the project is not maintained very very well. Last updated 2013-04-30... But I'll try to.

**Comment by nschlia** [comment]
All done (upstream fixes later).

Thanks for the test binary. Helped me a lot :)

**Comment by starius** [comment]
Some files are still lacking `\n`:

**Comment by starius** [comment]
I added test commands for libdvdetect and libcddb. It revealed some problems with pc files, which I fixed.

To apply my fixes, do the following:
```
wget -O fixes.patch https://gist.githubusercontent.com/starius/1355b2714178baf785616ea1de0f8040/raw/7f9b4238154be76bc3561b91a671160bbfeca085/-
git am fixes.patch
```

After applying this patch, everything looks good to me.

**Comment by nschlia** [comment]
Patch applied. Thanks for helping me out.

**Comment by nschlia** [comment]
Everything complete now.

File: src/libcddb-1-fixes.patch (hunk 12 above)
Comment (nschlia) [inline src/libcddb-1-fixes.patch]: Created ticket for the upstream project:

https://sourceforge.net/p/libcddb/bugs/11/

**Comment by starius** [comment]
One more thing. I think, we should squash all commits to one, because most of them are fixes of previous commits. Are you Ok with me doing it and pushing as [single commit](https://gist.github.com/f48006fa64c875be347ef5ea92baaa98)? Or you can do it manually by running `git rebase --interactive` and `git push -f`.

**Comment by pavelvat** [comment]
problem https://github.com/mxe/mxe/pull/1591#issuecomment-267466757 is still present in file neon-test.c 

>I think, we should squash all commits to one

Maybe to three commits: libcddb, libdvdetect, neon.

**Comment by starius** [comment]
> Maybe to three commits: libcddb, libdvdetect, neon.

+1

File: src/neon.mk [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+    $(INSTALL) -d '$(PREFIX)/$(TARGET)/lib/pkgconfig'
+    (echo 'prefix=$(PREFIX)/$(TARGET)'; \
+     echo 'exec_prefix=$${prefix}'; \
+     echo 'libdir=$${exec_prefix}/lib'; \
+     echo 'includedir=$${prefix}/include'; \
+     echo ''; \
+     echo 'Name: $(PKG)'; \
+     echo 'Version: $($(PKG)_VERSION)'; \
+     echo 'Description: neon is an HTTP and WebDAV client library'; \
+     echo 'Requires.private: openssl gettext'; \
```
Comment (starius) [inline src/neon.mk]: There is no such pc file (gettext.pc).
It builds without gettext. Why have you added gettext?

**Comment by starius** [comment]
As 3 commits: https://gist.github.com/bd33cc67ec9f8d49737bdc9f45c657bc
(gettext not included)

File: src/neon.mk (hunk 19 above)
Comment (nschlia) [inline src/neon.mk]: Because it uses libintl which is part of the gettext lib. The static version only links for me when its present.

Comment (starius) [inline src/neon.mk]: For me it complains as follows:

```
# create test binary
i686-w64-mingw32.static-gcc -W -Wall -Werror -std=c11 '/home/mxe/mxe-libcddb/src/neon-test.c' -o '/home/mxe/mxe-libcddb/usr/i686-w64-mingw32.static/bi
n/test-neon.exe' `i686-w64-mingw32.static-pkg-config neon --cflags --libs`
Package gettext was not found in the pkg-config search path.
Perhaps you should add the directory containing `gettext.pc'
to the PKG_CONFIG_PATH environment variable
Package 'gettext', required by 'neon', not found
/tmp/ccyTnSiE.o:neon-test.c:(.text+0x21): undefined reference to `ne_addr_resolve'
/tmp/ccyTnSiE.o:neon-test.c:(.text+0x37): undefined reference to `ne_addr_result'
/tmp/ccyTnSiE.o:neon-test.c:(.text+0x5a): undefined reference to `ne_addr_error'
/tmp/ccyTnSiE.o:neon-test.c:(.text+0x87): undefined reference to `ne_addr_first'
/tmp/ccyTnSiE.o:neon-test.c:(.text+0xaf): undefined reference to `ne_iaddr_print'
/tmp/ccyTnSiE.o:neon-test.c:(.text+0xce): undefined reference to `ne_addr_next'
/tmp/ccyTnSiE.o:neon-test.c:(.text+0xfa): undefined reference to `ne_addr_destroy'
collect2: error: ld returned 1 exit status
```

Can you show linking errors which you get without adding gettext, please?

Comment (nschlia) [inline src/neon.mk]: [neon_i686-w64-mingw32.static.txt](https://github.com/mxe/mxe/files/658744/neon_i686-w64-mingw32.static.txt)
[neon_x86_64-w64-mingw32.static.txt](https://github.com/mxe/mxe/files/658745/neon_x86_64-w64-mingw32.static.txt)

This strange, with gettext all neon builds fail. Without I get this (see attached full logs):

/opt/mxe/usr/x86_64-w64-mingw32.static/lib/libneon.a(ne_socket.o):/opt/mxe/tmp-neon-x86_64-w64-mingw32.static/neon-0.30.2/src/ne_socket.c:875: more undefined references to `libintl_dgettext' follow
/opt/mxe/usr/x86_64-w64-mingw32.static/lib/libneon.a(ne_openssl.o): In function `make_chain':
/opt/mxe/tmp-neon-x86_64-w64-mingw32.static/neon-0.30.2/src/ne_openssl.c:435: undefined reference to `libintl_fprintf'

Comment (nschlia) [inline src/neon.mk]: I manually rebuilt gettext (by deleting the installed/gettext file) but neon static still does not build. I am currently building the static variants, this is not yet complete, but I don't see why this should be a problem.

Comment (starius) [inline src/neon.mk]: neon makes use of gettext, if it detects its presence:

from [my log](https://gist.githubusercontent.com/starius/9510d615226bdb8123d034fe4a6a433b/raw/a560beb1132287166ea7a1c7df0fed9587c913fd/neon_i686-w64-mingw32.static):
```
checking for libintl.h... no
```
from [your log](https://github.com/mxe/mxe/files/658744/neon_i686-w64-mingw32.static.txt):
```
checking for libintl.h... yes
```

If it detects gettext, neon's code depends on symbols from gettext, causing linking errors if gettext is not passed to a linker. This is an example of [undeclared dependency](https://github.com/mxe/mxe/issues/1111).

To fix it, gettext should be declared as a dependency and passed to a linker. Unfortunately there is no gettext pc file, so adding it to `Requires.private` does not work. We should pass it to `Libs.private`. gettext itself depends on iconv, so iconv should also be passed to a linker. See my changes in the patch: https://gist.github.com/57e9804f504ecbc12efd7f9e23f48e39 It works here. Can you check it and confirm merging this pull request as [three commits](https://gist.github.com/82a402741d68c0f800b22aa1d28aeeae), please?

Comment (nschlia) [inline src/neon.mk]: Now I see, it was the test executable that failed. With your patch all is OK:

[build]     neon                     i686-w64-mingw32.static
[done]      neon                     i686-w64-mingw32.static                              6852 KiB       1m39.173s
[build]     neon                     i686-w64-mingw32.shared
[done]      neon                     i686-w64-mingw32.shared                              7036 KiB       1m38.384s
[build]     neon                     x86_64-w64-mingw32.static
[done]      neon                     x86_64-w64-mingw32.static                            7164 KiB       2m6.942s
[build]     neon                     x86_64-w64-mingw32.shared
[done]      neon                     x86_64-w64-mingw32.shared                            7348 KiB       2m14.793s

**Comment by starius** [comment]
I opened new pull request with squashed commits: https://github.com/mxe/mxe/pull/1598

**Pull request closed without merging**